rsksmart/tokenbridge
# Federator NFT

### Add Federator for reading NFT cross Event

Besides some config files needed to work with typescript, this is the first service in the federator for the NFT bridge

### Description

- Modified cross event variable names
- Add `FederatorNFT.ts` to the federator directory

### How to Test
- Actually, it's not possible to test it yet, but you can ensure that the tests for the main bridge are working properly ;) 

- Run both `ganache-cli`
- Enter into the federator directory and run the integration test 

#### Case 1

1. Go to bridge directory
```shell
$~ cd bridge
```

2. Run the first ganache-cli
```shell
$~ npm run ganache
```

3. Open another shell and run the ganache mirror
```shell
$~ npm run ganache-mirror
```

4. Compile and deploy the contracts to the ganache chains
```shell
$~ npm run deployLocalIntegrationTest
```

5. Go to the Federator directory and run the test for the main bridge federator
```shell
$~ cd ../federator
$~ npm run test
```

__Expected Result__
- It should pass the tests
![image](https://user-images.githubusercontent.com/17556614/130695320-b4635f64-35e1-4c08-891d-fa0379a2e932.png)


### Checklist

#### Bridge Directory `cd bridge`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Contracts are compiling `npm run compile`

#### Federator Directory `cd federator`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Typescript is compiling `npm run build`
